stl/inc/format

```diff
@@ -2761,27 +2794,89 @@ consteval typename _ParseContext::iterator _Compile_time_parse_format_specs(_Par
     using _FormattedType = conditional_t<is_same_v<_FormattedTypeMapping, typename basic_format_arg<_Context>::handle>,
         _Ty, _FormattedTypeMapping>;
     formatter<_FormattedType, _CharT> _Formatter{};
-    return _Formatter.parse(_Pc);
+    auto _Iter = _Formatter.parse(_Pc);
+    if constexpr (_Derived_from_formatter_base<formatter<_FormattedType, _CharT>>) {
+        return {_Iter, _Formatter._Specs};
+    } else {
+        return {_Iter, {}};
+    }
 }
 
 // set of format parsing actions that only checks for validity
 template <class _CharT, class... _Args>
 struct _Format_checker {
     using _ParseContext = basic_format_parse_context<_CharT>;
-    using _ParseFunc    = typename _ParseContext::iterator (*)(_ParseContext&);
+    using _ParseFunc    = pair<typename _ParseContext::iterator, _Dynamic_format_specs<_CharT>> (*)(_ParseContext&);
 
     static constexpr size_t _Num_args = sizeof...(_Args);
     _ParseContext _Parse_context;
     _ParseFunc _Parse_funcs[_Num_args > 0 ? _Num_args : 1];
+    size_t _Arg_use_count[_Num_args > 0 ? _Num_args : 1]{}; // only used for strings
+    bool _Is_string_type[_Num_args > 0 ? _Num_args : 1];
+    bool _Is_nullterminated_string_type[_Num_args > 0 ? _Num_args : 1];

```

Comment:
`_Num_args > 0 ? _Num_args : 1` occurs 4 times here - is it worth extracting to a named constant?